Use trace-level logging for logging due to SPI polling.

hsudbrock opened the pull request:
Resolves #343 

As discussed in https://github.com/zsmartsystems/com.zsmartsystems.zigbee/issues/343#issuecomment-408442532, this PR switches from DEBUG-logging to TRACE-logging for log messages caused by SPI polling.

In `SpiFrameHandler#processSpiCommand`, we need to find out whether the received frame is a `NoCallbacksResponse`. I use `EzspFrame#createHandler` here for getting the `EzspFrame` from the frame bytes. An alternative would be to inspect only the frame ID byte - checking whether it is `0x07` which encodes the `NoCallbacksResponse`. I decided against this, as the current solution will keep the EZSP-related logic in the `EzspFrame` class - but please comment if you think that the alternative is preferable.

cdjackson commented on com.zsmartsystems.zigbee.dongle.ember/src/main/java/com/zsmartsystems/zigbee/dongle/ember/internal/spi/SpiFrameHandler.java:
```diff
@@ -311,11 +311,20 @@ private void processSpiCommand(int[] packetData) {
+                
+                // Only trace-log in case of NoCallBacksResponse (which is due to polling)
+                String logMessage = String.format("<-- RX SPI frame: %s", frameToString(packetData));
+                if (EzspFrame.createHandler(Arrays.copyOfRange(packetData, 2, packetData.length)) instanceof EzspNoCallbacksResponse) {
```
Instead of this call to ```createHandler``` I wonder if it might not be "better" to peek into the packet and check the appropriate byte for the CallBack Response ID? I know that's not nice as it's breaking the layers, but...
Alternatively, I wonder if we shouldn't copy the data into a temporary buffer to avoid multiple calls to ```copyOfRange```?

hsudbrock replied:
I had also thought about that, and I cannot really say if either of the two solutions is "better". I will replace by code that "peek into the packet", just to have it explicit (we can still go back if you have some argument that we should not do the "peeking").

Peeking requires to hard-code the frame ID for the NoCallbacksResponse  (`0x07`) here, as the autocoder-generated frame ID constants in the `EzspFrame` class are `protected`.

hsudbrock commented on com.zsmartsystems.zigbee.dongle.ember/src/main/java/com/zsmartsystems/zigbee/dongle/ember/ZigBeeDongleEzsp.java:
```diff
@@ -404,7 +405,13 @@ public void setZigBeeTransportReceive(ZigBeeTransportReceive zigbeeTransportRece
-        logger.debug("RX: " + response.toString());
+        // Only trace-log in case of NoCallbacksResponse (which is due to SPI polling)
+        String logMessage = "RX: " + response.toString();
+        if (response instanceof EzspNoCallbacksResponse) {
```
I agree, this could be contained in the `SpiFrameHandler` as you describe it here. I will change.

hsudbrock commented on com.zsmartsystems.zigbee.dongle.ember/src/main/java/com/zsmartsystems/zigbee/dongle/ember/ZigBeeDongleEzsp.java:
```diff
@@ -479,6 +486,11 @@ public void handlePacket(EzspFrame response) {
             return;
         }
+        
+        if (response instanceof EzspNoCallbacksResponse) {
```
Agree.

hsudbrock commented:
I pushed changes addressing your comments; I also removed a 'double' call to `EzspFrame.createHandler(...)` for the same packet data in `SpiFrameHandler#processSpiEzsp` over which I stumbled when doing the changes.

hsudbrock commented:
I think you have to configure "format on save" explicitly in Eclipse - which I probably did not do :/
Let me explicitly trigger the formatter once more for the changed files and commit (I thought I did format this time - sorry if there is still something missing...)

hsudbrock commented:
I updated the PR.

> I hope it effectively achieves what you were originally wanting to do with the use of the extra logger...

There is still a lot of logging happening in some circumstances, even with the polling-induced logging removed. I will see whether this is still "too" bothering in some circumstances when working with the library - but I think we can leave as is for now, and can still re-start the discussion if working is impeded by too much logging that I cannot configure accordingly in the future.

Thanks for your support with this PR!
